ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -323,7 +323,11 @@ private void checkAndMaybeComplete() {
 
     if (operationCompleted) {
       Exception e = operationException.get();
+      if (operationResult == null && e == null) {
+        e = new RouterException("Operation failed, but exception was not set", RouterErrorCode.UnexpectedInternalError);
```

Comment:
But don't you want to know when it happens?  I think this would be a programming error - a flow that has been missed.